Client/Frontend/Browser/Tab.swift

```diff
@@ -275,12 +286,8 @@ class Tab: NSObject {
     deinit {
         deleteWebView()
         contentScriptManager.helpers.removeAll()
-        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
-        
-        let rewards = appDelegate.browserViewController.rewards
-        
         if !PrivateBrowsingManager.shared.isPrivateBrowsing {
-            rewards.reportTabClosed(tabId: rewardsId)
+            self.rewards?.reportTabClosed(tabId: rewardsId)
```

Comment:
I'll handle this at a later date, but we should remove this entirely (and thus Tab's dependency on the `BraveRewards` object)